### ISSUE TITLE
Opening Roslyn with Update 3 gets Stack Overflow in GetNetExportedType

### ISSUE THREAD

**Issue by paul1956** [dotnet/roslyn#12406]
**Version Used**: 
Update 3
**Steps to Reproduce**:
1. Open Visual Studio to debug VSIX, hit F5, Open Roslyn.sln or Open Roslyn.sln

**Expected Behavior**:
Solution should open
**Actual Behavior**:
Stack Overflow in GetNetExportedType

```
System.StackOverflowException was unhandled
Message: An unhandled exception of type 'System.StackOverflowException' occurred in Microsoft.VisualStudio.Shell.Design.dll
```

**Comment by Pilchie** [comment]
Can you get us a crash dump, or at least a call stack? This isn't something anyone here is seeing.

**Comment by tmat** [comment]
My guess is that this is a known bug in LMR filed internally.

**Comment by paul1956** [comment]
I can reproduce the issue at will but can't get the Frown to capture it. Let me know if you still need it or if you can reproduce it have an ETA to get it fixed.

**Comment by Pilchie** [comment]
@tmat Do you have more context on the internal bug?

**Comment by Pilchie** [comment]
Note - I hit this over the weekend too.  Moving to compilers, since that's where the internal bug is.  @jaredpar - can we get someone to look at that?

**Comment by jaredpar** [comment]
@Pilchie do you have a full stack trace of the failure?

**Comment by Pilchie** [comment]
I have a machine that seems to repro this every time I open Roslyn.sln now if you need an environment to debug.

Repeating portion of stack:

>Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.AssemblyNameHelper.AssemblyNameFromRefBuilder.Fetch()	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.AssemblyNameHelper.AssemblyNameBuilder.CalculateName()	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.AssemblyNameHelper.GetAssemblyNameFromRef(System.Reflection.Adds.Token assemblyRefToken, Microsoft.MetadataReader.MetadataOnlyModule module, Microsoft.MetadataReader.IMetadataAssemblyImport assemblyImport)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.PseudoCustomAttributes.GetRawTypeForwardedToAttribute(Microsoft.MetadataReader.MetadataOnlyModule manifestModule, string fullname, bool ignoreCase)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.MetadataExtensionsPolicy20.TryTypeForwardResolution(Microsoft.MetadataReader.MetadataOnlyAssembly assembly, string fullname, bool ignoreCase)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.MetadataOnlyAssembly.GetType(string name, bool throwOnError, bool ignoreCase)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!System.Reflection.Adds.TypeNameParser.ParseTypeName.AnonymousMethod__1(System.Reflection.Assembly assembly, string simpleTypeName, bool ignoreCase)	Unknown
 	mscorlib.dll!System.TypeNameParser.ResolveType(System.Reflection.Assembly assembly, string[] names, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.TypeNameParser.ConstructType(System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.TypeNameParser.GetType(string typeName, System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.Type.GetType(string typeName, System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!System.Reflection.Adds.TypeNameParser.ParseTypeName(System.Reflection.Adds.ITypeUniverse universe, System.Reflection.Module module, string input, bool throwOnError)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.UnresolvedTypeName.ConvertToType(System.Reflection.Adds.ITypeUniverse universe, System.Reflection.Module moduleContext)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.MetadataExtensionsPolicy20.TryTypeForwardResolution(Microsoft.MetadataReader.MetadataOnlyAssembly assembly, string fullname, bool ignoreCase)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!Microsoft.MetadataReader.MetadataOnlyAssembly.GetType(string name, bool throwOnError, bool ignoreCase)	Unknown
 	Microsoft.VisualStudio.Shell.Design.dll!System.Reflection.Adds.TypeNameParser.ParseTypeName.AnonymousMethod__1(System.Reflection.Assembly assembly, string simpleTypeName, bool ignoreCase)	Unknown
 	mscorlib.dll!System.TypeNameParser.ResolveType(System.Reflection.Assembly assembly, string[] names, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.TypeNameParser.ConstructType(System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.TypeNameParser.GetType(string typeName, System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError, bool ignoreCase, ref System.Threading.StackCrawlMark stackMark)	Unknown
 	mscorlib.dll!System.Type.GetType(string typeName, System.Func<System.Reflection.AssemblyName, System.Reflection.Assembly> assemblyResolver, System.Func<System.Reflection.Assembly, string, bool, System.Type> typeResolver, bool throwOnError)	Unknown

**Comment by jaredpar** [comment]
This doesn't look like a LMR bug.  It looks like DTAR stack overflowing again.

**Comment by paul1956** [comment]
@jaredpar with the TLA's (Three Letter Acronyms)  does that mean it is assigned to someone to fix? This happens with VS 15 as well.

**Comment by jaredpar** [comment]
@paul1956 we're trying to route to the appropriate team.  The only crash we've been able to repro though is on Dev15 (not VS2015 / Dev14).  Are you seeing it on Dev14?  

As for the TLAs:

- DTAR: design time assembly resolution.  Part of the Visual Studio service for helping resolve references in the IDE space. 
- LMR: lightweight metadata reader.  A managed wrapper around the old native metadata import interfaces.

**Comment by paul1956** [comment]
Both but 100% of the time on Dev15 and rarely and randomly on Dev14, On Dev14 I can't reliably reproduce and never when sending a Frown so that it can be captured. The only consistent thing is the error message "Stack Overflow in GetNetExportedType".

**Comment by jaredpar** [comment]
@jasonmalinowski this looks like another DTAR stack ovterflow issue.  Do you know where to route this bug?

**Comment by jaredpar** [comment]
This ended up getting reported through another channel and developers involved were able to get a consistent repro on Dev14 Update 3.  It does look to be outside the Roslyn components.  But given the primary bug is internal now, going to leave this open as a tracking issue until the internal one is resolved.

Context: https://devdiv.visualstudio.com/DevDiv/_workitems?id=249615&_a=edit

**Comment by jaredpar** [comment]
@rchande assigning to you based on the internal email.

**Comment by Pilchie** [comment]
@rchande didn't you check something in to address this internally?

**Comment by CyrusNajmabadi** [comment]
Can't repro this now.  Based on internal vs tracking, this was fixed in: in VS 2017 RC 15.0.0-RC.4+26026.0 in August 2017.  So closing out.
